zict: disable python 2 and 3.5.

teh opened the pull request:
Part of ZHF in #97479

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Zero hydra failures

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

drewrisinger commented on pkgs/development/python-modules/distributed/default.nix:
```diff
@@ -35,11 +36,14 @@ buildPythonPackage rec {
   };
 
+  # zict dependency is now python 3.6+ only
+  disabled = !(pythonAtLeast "3.6");
```
```suggestion
  disabled = pythonOlder "3.6";
```

drewrisinger commented on pkgs/development/python-modules/streamz/default.nix:
```diff
@@ -38,6 +39,9 @@ buildPythonPackage rec {
   ];
 
+  # zict dependency is now python 3.6+ only
+  disabled = !(pythonAtLeast "3.6");
```
```suggestion
  disabled = pythonOlder "3.6";
```

drewrisinger commented on pkgs/development/python-modules/streamz/default.nix:
```diff
@@ -38,6 +39,9 @@ buildPythonPackage rec {
     requests
   ];
 
+  # zict dependency is now python 3.6+ only
```
Wording is a little confusing, implies that it would support e.g. py3.5 if not for zict. Maybe just say is python 3.6+ only, though that's implied by ``disabled=pythonOlder 3.6``

drewrisinger reviewed (changes requested):
* [x] Diff LGTM, minus change to clearer ``pythonOlder``
* [ ] commits should be broken up, one per file.
* [x] builds via ``nix-review``

```
1 package removed:
python2.7-zict (†2.0.0)
```

jonringer commented on pkgs/development/python-modules/streamz/default.nix:
```diff
@@ -38,6 +39,9 @@ buildPythonPackage rec {
     requests
   ];
 
+  # zict dependency is now python 3.6+ only
```
then dict should be marked accordingly, I wouldn't mark this as disabled just because a dependency doesn't work on it.

One example would be if someone override zict, streamz should still work.

teh replied:
Thanks for the review! To clarify: You are saying that marking zict as `pythonOlder "3.6";` is enough?

teh replied:
Ok thanks - updated accordingly.

drewrisinger commented:
Can you break this into 3 commits, one per file? You can do this with ``git reset --soft HEAD~1``, then individually commit & force-push ``git push -f``.

PR & commit titles should also be named something like ``python27Packages.zict: disable`` (NOTE the pythonPackages prefix especially).

jonringer reviewed (approved):
LGTM

Result of nixpkgs-review pr 97605 [1](https://github.com/Mic92/nixpkgs-review)

jonringer reviewed (changes requested):
oh, diff lgtm, but commit history should be split into 3, as said above

teh committed:
zict: disable build for Python older than 3.6

No longer supported as of
https://github.com/dask/zict/commit/4d5e14dadba940235a59f7e5076653cf7d51d27d

teh committed:
streamz: disable build for Python older than 3.6

teh committed:
distributed: disable build for Python older than 3.6
